### DISCUSSION TITLE
Ansprechverhalten

### DISCUSSION THREAD

**Discussion by cyberopi** [brutella/hkknx-public#261] [Q&A] [answered]
Wie ist die Geschwindigkeit bei der Aktualisierung der KNX Geräte bei Neustart von Homekit?
Bei sagen wir mal 80 Geräten?

**Comment by chef76** [comment, marked as answer]
Meine 113 Geräte brauchen ca. 2 Minuten. Man sieht ja jetzt unter Status, wie die hochgezählt werden. Unter 2.4 ging das noch schneller, aber da bremst jetzt anscheinend die Telegrammbeschränkung ein kleines bisschen aus.

**Comment by cyberopi** [comment]
Das heißt jedes Mal wenn ich in die App gehe dauert es 2Min bis ich zum Beispiel den Status einer Lampe sehe?

**Reply by dreamdatcrap** [reply]
Hi,

nein, da hast du glaube ich ein kleinen Verständnisdreher :)

Deine HomeKit Brücke wird wohl eine kurze Zeit brauchen um sich zu initialisieren. Das hat aber nichts mit deiner Home App auf dem iPhone zu tun. Ist die Brücke verfügbar, zeigt die Home App den Status direkt an.

gruß

**Reply by chef76** [reply]
Wenn du die App öffnest, dauert es so ca. 2 Sekunden, bis alle Stati verfügbar sind. Nur bei einem Neustart der Brücke (vor meinem letzten Update lief sie 236 Tage ohne Probleme) dauert es die 2 Minuten!

**Comment by cyberopi** [comment]
Ok Danke für die Anwort ich habe gerade den Prosrv da dauert es sehr lange bis zu garnicht bis Anwort kommt. Liegt hier anscheinend an Homekit ??? sagte man mir und ich soll weniger Geräte anlegen.
Dann werde ichHKKNX mal testen. läuft die software auch im Standby Modus vom MAC M1?

**Comment by cyberopi** [comment]
Gibt es für den Mac eine Installationsanleitung ? Den Download habe ich gefunden hkknx-2.5.1_linux_arm.tar.gz 
Die Dateien im gepackten Order lassen sich aber nicht ausführen. DIe MD verlangt nach einem Programm

**Reply by chef76** [reply]
Es geht auf dem Mac aber nicht im Standby. Probiert hab ich es noch nicht, aber das ist auch keine gute Lösung, was ich so bisher in den Foren gelesen hab. Ich habs auf einem Raspi 3 laufen: https://hochgatterer.me/de/hkknx/docs/install/rpi-img/

**Reply by brutella** [reply]
Läuft auf dem Mac ohne Probleme. Du musst aber die richtige Datei herunterladen. Für eine M1 Mac ist das  hkknx-x.x.x_**darwin_arm64**
https://hochgatterer.me/de/hkknx/docs/install/downloads/

Danach kannst du die Datei testweise in Terminal.app ausführen. `./hkknx -db ./database`
https://hochgatterer.me/de/hkknx/docs/install/rpi-cli/#installation

Wenn du hkknx ständig laufen lassen willst, kannst du das auf dem Mac mit [launchd](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html) machen.

Im folgenden Beispiel wird der Ordner `/Users/{user}/Library/Application Support/at.mah.hkknx` verwendet – du kannst aber eine anderen Ordern wählen. Statt `{user}` musst du deinen Benutzernamen eintragen.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Label</key>
	<string>at.mah.hkknx</string>
	<key>ProgramArguments</key>
	<array>
		<string>/Users/{user}/Library/Application Support/at.mah.hkknx/hkknx</string>
		<string>--db</string>
		<string>/Users/{user}/Library/Application Support/at.mah.hkknx/db</string>
		<string>--verbose</string>
	</array>
	<key>StandardErrorPath</key>
	<string>/Users/{user}/Library/Application Support/at.mah.hkknx/stderr.log</string>
	<key>StandardOutPath</key>
	<string>/Users/{user}/Library/Application Support/at.mah.hkknx/stdout.log</string>
    <key>KeepAlive</key>
    <true/>
</dict>
</plist>
```

Mit diesen launchd Daemon wird hkknx bei einem Neustart von deinem Mac automatisch gestartet bzw. bei einem Programmabbruch automatisch wiedergestartet.

**Comment by cyberopi** [comment]
Hab die Testversion auf einem Raspi hingekriegt :-)
eine Frage warum funkt der Raspi ins internet?

**Reply by brutella** [reply]
Welchen Traffic meinst du genau?

**Reply by cyberopi** [reply]
Ich hatte ein Internet Symbol in der Fritzbox gesehen . Das ist jetzt aber weg . Vielleicht habe ich mich verguckt
LG
